packages/backend/src/models/ValidationModel/ValidationModel.ts

```diff
@@ -38,10 +42,28 @@ export class ValidationModel {
         await this.database.transaction(async (trx) => {
             const insertPromises = validations.map((validation) =>
                 trx(ValidationTableName).insert({
-                    saved_chart_uuid: validation.chartUuid,
-                    dashboard_uuid: validation.dashboardUuid,
+                    saved_chart_uuid: isChartValidationError(validation)
```

Comment:
why do you need to do this check ? isChartValidationError

Reply:
oh I see, you created new types with different fields, in that case, how about creating a differnet `insertChart / insertDashboard` methods with the right CreateChatValidation as argument, so you don't need to do all these checks

Reply:
Yea, the new types make it clearer on what gets inserted, so having those type checks is necessary. I've take your suggestions here and here: https://github.com/lightdash/lightdash/pull/5682#discussion_r1211372953, and refactored this to make it cleaner. LMK what you think 😄 https://github.com/lightdash/lightdash/pull/5682/commits/6718f6bcd45bf37bb6ecadc6a139d5840f35de57